modules/consentManagementUsp.js

```diff
@@ -0,0 +1,309 @@
+/**
+ * This module adds USPAPI (CCPA) consentManagement support to prebid.js. It
+ * interacts with supported USP Consent APIs to grab the user's consent
+ * information and make it available for any USP (CCPA) supported adapters to
+ * read/pass this information to their system.
+ */
+import * as utils from '../src/utils';
+import { config } from '../src/config';
+import { uspDataHandler } from '../src/adapterManager';
+import includes from 'core-js/library/fn/array/includes';
+
+const DEFAULT_CONSENT_API = 'uspapi';
+const DEFAULT_CONSENT_TIMEOUT = 50;
+const DEFAULT_ALLOW_AUCTION_WO_CONSENT = true;
+
+export let userUSP;
+export let consentTimeout;
+export let allowAuction;
+
+let consentData;
+let addedConsentHook = false;
+
+// consent APIs
+const uspCallMap = {
+  'uspapi': lookupUpsConsent
+};
+
+/**
+ * This function handles interacting with an USP compliant consent manager to obtain the consent information of the user.
+ * Given the async nature of the USP's API, we pass in acting success/error callback functions to exit this function
+ * based on the appropriate result.
+ * @param {function(string)} uspSuccess acts as a success callback when USPAPI returns a value; pass along consentObject (string) from UPSAPI
+ * @param {function(string)} uspError acts as an error callback while interacting with USPAPI; pass along an error message (string)
+ * @param {object} hookConfig contains module related variables (see comment in requestBidsHook function)
+ */
+function lookupUpsConsent(uspSuccess, uspError, hookConfig) {
+  function handleUspApiResponseCallbacks() {
+    const uspResponse = {};
+
+    function afterEach() {
+      if (uspResponse.usPrivacy) {
+        uspSuccess(uspResponse, hookConfig);
+      }
```

Comment:
this only moves on if there's no `usPrivacy` param.  otherwise it's left waiting for the timeout to finish.  If the callback is called, we need to move on whether we have a valid response or not.

Reply:
should be fixed now.